source/demo-function/src/handler.rb

```diff
@@ -0,0 +1,28 @@
+require 'aws-sdk-s3'
+require 'json'
+require 'logger'
+
+module Identity
+  module Idp
+    module Functions
+      module DemoFunction
+        class Handler
```

Comment:
- This is part of "the ruby way™" namespacing...it's a lot
- We could also make these more like mini gems and use `lib/handler` instead of `src/handler`?

Reply:
I think this is a good direction. Could `DemoFunction` be the class instead of having a `Handler` class?

Reply:
Sure! So the issue is, I wanted to be able to automatically load the entry point for each function here:

https://github.com/18F/identity-idp-functions/pull/2/files#diff-ba8a2d9b7e87f69cf3fa535486d6b84eR6-R10

My approach was to make `src/handler.rb` the entry point, hence why I called it handler.

So what do you think if we had a convention that the entry point is named after the function, so if our function is `demo-function`, then `source/demo-function/src/demo-function.rb`
```

So maybe we switch to underscored function names? `source/demo_function/src/demo_function.rb`

Reply:
That sounds good to me. Big agree on switching to underscores too.

Reply:
underscored in 77137d2a9f865ad6d2a9b01684f3a2b9239ca65d

Reply:
oof I just re-discovered our issue.

We have a `class Identity` model in the IDP, so we can't use `module Identity` in our gems.... I think this is why we went with all our other weird namespaces.

Any thoughts on what to name this? I'm not opposed to renaming the repo to have underscores, ex `identity_idp_functions` which would `IdentityIdpFunctions`

Reply:
Yep, I think that that's good

Reply:
gonna namespace without renaming the repo for now